herculens/MassModel/mass_model.py

```diff
@@ -99,14 +126,14 @@ def hessian(self, x, y, kwargs, k=None):
         :param k: only evaluate the k-th lens model
         :return: f_xx, f_xy, f_yx, f_yy components
         """
-        x = np.array(x, dtype=float)
-        y = np.array(y, dtype=float)
+        # x = np.array(x, dtype=float)
+        # y = np.array(y, dtype=float)
         if isinstance(k, int):
             f_xx, f_yy, f_xy = self.func_list[k].hessian(x, y, **kwargs[k])
             return f_xx, f_xy, f_xy, f_yy
 
         bool_list = self._bool_list(k)
-        f_xx, f_yy, f_xy = np.zeros_like(x), np.zeros_like(x), np.zeros_like(x)
+        f_xx, f_yy, f_xy = 0., 0., 0.
```

Comment:
this is a code comment: if x is an array the return object is not in the same format/dimension